content/blog/2021-10-05-extending-models-with-more-data.md

```diff
@@ -0,0 +1,242 @@
+descriptionLong: |
+  When you have an existing model trained for one problem, you might want to extend it to handle other problems. When you have
+  data versioning, it's easier to see which data additions make
+  your model better or worse and then you can see where to make
+  improvements.
+picture: 2021-10-05/cats-and-dogs.png
+pictureComment: Adding more data to your dataset for a more generic model
+author: milecia_mcgregor
+commentsUrl: https://discuss.dvc.org/t/extending-models-with-more-data/881
+tags:
+  - MLOps
+  - DVC
+  - Git
+  - Experiments
+  - Data Versioning
+---
+
+## Intro
+
+You might be in the middle of training a model and then the business problem
+shifts. Now you have this model that has been going through the training process
+with a specific dataset and you need to make the model more generic.
+
+There's likely something that your model learned that can be useful on this new
+dataset, so you might not have to restart the entire training process. We'll do
+an example of updating a pre-trained model to use a broader dataset with DVC. By
+the end of this, you should see how you can handle this quickly and start
+running new experiments to get a more generic model.
+
+## The original pre-trained model
+
+For this post, we'll be making a more generic image classifier by taking the
+original dataset with bees and ants and adding cats and dogs to it. You can
+clone [this GitHub repo](https://github.com/iterative/pretrained-model-demo) to
+get the current bees and ants model and check out
+[this post](https://dvc.org/blog/transfer-learning-experiments) on how we
+experimented with both AlexNet and SqueezeNet to build this model.
+
+So we're starting from our current bees and ants model and extending it to
+classify dogs and cats as well. We'll start by adding some cats and dogs data to
+our validation data and do some experiments with the current model to see how it
+performs on generic data.
+
+Then we'll add the cats and dogs data to the training data and watch how the
+model improves as we run experiments.
+
+## Updating the dataset with DVC
+
+To add the new cats and dogs dataset to the project, we'll use this DVC command.
+
+```dvc
+dvc get https://github.com/iterative/dataset-registry blog/cats-dogs
+```
+
+This downloads a sample dataset with images of cats and dogs. You can use this
+command to download files or directories that are tracked by DVC or Git. This
+command can be used from anywhere in the file system, as long as DVC is
+installed.
+
+This will make a new directory called `./cats-dogs/data/` that was downloaded
+from the DVC remote and it has images for cats and dogs. The data is in a
+similar format to the ants and bees directories so there aren't any other code
+changes we need to make.
+
+We do need to slowly add in the new data though. For starters, we'll move the
+`val` data for `cats` and `dogs` from the `/cats-dogs/data/` directory to the
+corresponding directory in `data/hymenoptera_data`.
```

Comment:
This is picky, but I think hymenoptera is the classification for bees and ants or something like that. It would probably be more natural to add cats and dogs to the base `data` directory, but that requires minor code and pipeline changes. Not sure if it's problematic to make changes from the previous blog post.

Reply:
I think if there's an explanation for moving the bees and ants data out of the `hymenoptera_data` folder into the `data` folder, that should be fine. Let me know what you think about the update.

Reply:
I probably shouldn't have said anything since it's a minor detail that has nothing to do with the technical content 😄 . 

The explanation and changes in the update look good, but it also feels like an unnecessary distraction. Maybe it's better to go back to how it was and just have a quick note, like "(cats and dogs don't really belong in the [hymenoptera](https://en.wikipedia.org/wiki/Hymenoptera) directory, but it's the easiest way to add the data)." 

Totally up to you here.

Reply:
I went back to the old version and added that little note.